res/controllers/midi-components-0.0.js

```diff
@@ -656,7 +656,7 @@
                 });
             }
 
-            _.merge(this, newLayer);
+            Object.assign(this, newLayer);
```

Comment:
I don't think there was ever really a need for a deep merge here, so I just replaced by `Object.assign()`.

Reply:
I'm a little skeptical about this. It is a breaking change, so let's double check this is actually true for at least the mappings we include.

Reply:
I already checked the current usages but a double check is always a good idea.

Reply:
yeah, so I also double checked and while it shouldn't cause any issues in-tree now it I'm still not sure if its a good decision from an API design standpoint. ComponentJS is essentially based on the ideas of QML, so I'd like to retain the same behavior as when you create QML components. It's essentially described as [grouped properties](https://doc.qt.io/qt-6/qtqml-syntax-objectattributes.html#grouped-properties) and when you transfer the behavior to JS, it does a deep object merge rather than an `Object.assign`. So I'd like to retain the similarity there and keep using the behavior of `_.merge`. I'm still in favor of removing lodash, but then we'd need an implementation of our own object merge. wdyt?

Reply:
Ok so I initially wrote a `deepAssign`implementation before checking if deep merge was necessary and deciding to drop it. I can bring back that implementation.

Reply:
Yes, please do so.

Reply:
Done. So the code I write is far simpler that Lodash's but should suffice in most cases. Here is a test sample:

```js
target = {
    a: "a",
    b: [1, 2],
    c: {
        d: {
            e: [1, 2]
        },
    },
    g: new Date()

}

source = {
    b: [2, 3, new Date()],
    c: {
        d: {
            f: new Date()
        }
    },
    g: {test: function(){}},
    h: function test() {}
}

Object.deepMerge(target, source);
// Should produce:
 {
    a: "a",
    b: [1, 2, 2, 3, new Date()],
    c: {
        d: {
            e: [1, 2],
            f: new Date(),
        },
    },
    g: {test: function(){}},
    h: function test() {}
}
```